### PR TITLE
fix: XSum with "2nd grader" replaced and BLUE added and naming changed

### PR DESCRIPTION
**Pull request by tianjianjiang**
- Mostly for https://github.com/bigscience-workshop/promptsource/pull/360#issuecomment-916532724
- Naming and some wording changed
  - Hopefully name changes may hint structural/orthographic differences
     1. ~Instructive punctuations `:` and `?`;~
     2. ~Custom separators `===` and `"""`~;
     3. Placeholder ~`<DOC>`~ `DOC`.

### CODE REVIEW

File: promptsource/templates/xsum/templates.yaml [hunk 1]
```diff
@@ -10,15 +10,16 @@ templates:
       ===
 
 
-      Given the above document, generate a summary: ||| {{summary}}'
+      Write a summary of the text above : ||| {{summary}}'
```
Comment (tianjianjiang) [inline promptsource/templates/xsum/templates.yaml:13]: I guess the author of this particular template wanted to compare between prompts with or without "one sentence." 
If that's the case, this one line change shall be rejected.

File: promptsource/templates/xsum/templates.yaml [hunk 2]
```diff
@@ -133,8 +139,9 @@ templates:
       choices_in_prompt: false
       metrics:
       - ROUGE
+      - BLEU
       original_task: true
-    name: write_one_sentence
+    name: '<DOC>===given_above_write_one_sentence:'
```
Comment (tianjianjiang) [inline promptsource/templates/xsum/templates.yaml:144]: cf.

**Review by awebson — CHANGES_REQUESTED** [review]
Thank you! Although could you please remove the <DOC>, etc. in template names? All punctuations will be converted to underscores, so just write a few words + whitespace/underscore is fine.

Also, could you replace triple quotes with regular quotes or just paragraph breaks? Triple quotes are not natural English.

@stephenbach Double checking: does streamlit truncate long text in rendering? In dataset viewer, some examples are clearly truncated, even though Tian-Jian simply wrote `{{document}}`.

**Comment by tianjianjiang** [comment]
@awebson 
> remove the , etc. in template names? All punctuations will be converted to underscores, so just write a few words + whitespace/underscore is fine.

I see, will do! (I thought the conversion has been done by streamlit 💦 )

> Also, could you replace triple quotes with regular quotes or just paragraph breaks? Triple quotes are not natural English.

Understood! How about `===` then? I was under the impression that some of those uncommon usages were allowed before such that some templates may gain a side effect of tricking the tokenizer and the final state.

**Comment by awebson** [comment]
`===` plausibly aids in human reading, and it's used by so many other templates that (a) too late to change now (b) less risk in being learned as a spurious feature of one particular task.

**Comment by tianjianjiang** [comment]
@awebson Got it, thanks!

**Comment by tianjianjiang** [comment]
@awebson
> `===` plausibly aids in human reading, and it's used by so many other templates that (a) too late to change now (b) less risk in being learned as a spurious feature of one particular task.

Sorry, just want to be certain. Does it still mean that I should also remove `===` from names? (I haven't checked seqio code for the conversion rules, my apologies.)

**Comment by awebson** [comment]
Yes, remove === from template names but leave them them in the jinja.

**Comment by tianjianjiang** [comment]
@awebson Roger that!

File: promptsource/templates/xsum/templates.yaml [hunk 3]
```diff
@@ -34,41 +35,43 @@ templates:
...
+    jinja: 'My college roommate asked me what this article means:
 
 
-      """ {{document}} """
+      {{document}}
 
 
-      I reorganized it in a simple sentence that even an elementary school student
-      can understand: ||| {{summary}}'
+      So I recap it in layman''s terms: ||| {{summary}}'
```
Comment (awebson) [inline promptsource/templates/xsum/templates.yaml:67]:
```suggestion
      So I recapped it in layman''s terms: ||| {{summary}}'
```

**Review by awebson — APPROVED** [review]
Thank you!

File: promptsource/templates/xsum/templates.yaml (hunk 3 above)
Comment (tianjianjiang) [reply promptsource/templates/xsum/templates.yaml]: Ah, thank you!

**Comment by tianjianjiang** [comment]
@awebson @stephenbach
> @stephenbach Double checking: is streamlit truncating long text in rendering? In dataset viewer, some examples are clearly truncated, even though Tian-Jian simply wrote `{{document}}`.

`template.apply()`'s default will truncate long strings. I'm gonna submit a simple PR for it.
